lib/EarlyFunctionAnalysis/AttachDebugInfo.cpp

```diff
@@ -149,8 +149,14 @@ class DebugInformationBuilder {
     DILocation *CurrentDI = DefaultDI;
 
     for (auto *BB : ReversePostOrderTraversal(&F)) {
-    if (not GCBI.isTranslated(BB))
+      if (not GCBI.isTranslated(BB)) {
+        // This is a dispatcher-related basic block, fill its debug information
+        // with function entry point to avoid leaving it empty.
+        for (auto &I : *BB)
+          I.setDebugLoc(DefaultDI);
+
         continue;
+      }
```

Comment:
This is doing more than you think it does.

`not GCBI.isTranslated(BB)` implies that the block type (see `BlockType`) is not `JumpTargetBlock` or `TranslatedBlock`.

I'd say most of the other options are not attachable to any meaningful debug location, therefore using a default (as you're doing is fine).

However, `IndirectBranchDispatcherHelperBlock` is a dispatcher introduced in the code to handle an indirect jump (`jmp rax`). So all of the body of the `SwitchInst` and the related blocks should probably be attached to the debug location of the instruction of the jump instruction.

Reply:
Sounds good!

Reply:
Done, please take a fresh look.